security/leaks.js

```diff
@@ -0,0 +1,140 @@
+                            <code>${JSON.stringify(actual[k], null, 2)}</code>
+                        </details>
+                    `, 'added');
+                    return;
+                }
+                const changed = ['configurable', 'enumerable', 'writable', 'type', 'value'].filter(attr => actual[k][attr] !== exp[attr]);
+                if (changed.length === 1) {
+                    addBullet(`
+                        <details>
+                            <summary><code>${breadcrumbs.join('.')}.${k}</code> changed ${changed.join(', ')}</summary>
+                            <ul>
+                                <li>Expected: <code>${JSON.stringify(exp[changed[0]], null, 2)}</code></li>
+                                <li>Actual: <code>${JSON.stringify(actual[k][changed[0]], null, 2)}</code></li>
+                            </ul>
+                        </details>`, 'changed');
+                } else if (changed.length > 0) {
+                    addBullet(`
+                        <details>
+                            <summary><code>${breadcrumbs.join('.')}.${k}</code> changed ${changed.join(', ')}</summary>
+                            <ul>
+                                <li>Expected: <code>${JSON.stringify(exp, null, 2)}</code></li>
+                                <li>Actual: <code>${JSON.stringify(actual[k], null, 2)}</code></li>
+                            </ul>
+                        </details>`, 'changed');
+                }
+
+                if (actual[k].properties) {
+                    if (!exp.properties) {
+                        addBullet(`
+                        <details>
+                            <summary><code>${breadcrumbs.join('.')}.${k}</code> changed</summary>
+                            <ul>
+                                <li>Expected: <code>${JSON.stringify(exp, null, 2)}</code></li>
+                                <li>Actual: <code>${JSON.stringify(actual[k], null, 2)}</code></li>
+                            </ul>
+                        </details>`, 'changed');
+                    } else {
+                        compareObjects(exp.properties, actual[k].properties, [
+                            ...breadcrumbs,
+                            k
+                        ]);
+                    }
+                }
+            });
+            Object.keys(expected).forEach((k) => {
+                if (!actual[k]) {
+                    addBullet(`
+                        <details>
+                            <summary>mising property <code>${breadcrumbs.join('.')}.${k}</code></summary>
+                            <code>${JSON.stringify(expected[k], null, 2)}</code>
+                        </details>
+                    `, 'removed');
+                }
+            });
+        }
+
+        compareObjects(platformExpected.window, window.collectedProps.window, ['window']);
+    }
+
+    document.querySelector('#run').addEventListener('click', run);
```

Comment:
Let's make this test runnable automatically:
- e.g. when called with ?run&browser=safari_14 it should do a comparison with safari_14
- let's write results to window.results like in other tests and make it downloadable as a JSON via "Download results" button